addons/account/models/account_report.py

```diff
@@ -296,6 +296,10 @@ def _validate_groupby_no_child(self):
             if report_line.parent_id.groupby:
                 raise ValidationError(_("A line cannot have both children and a groupby value (line '%s').", report_line.parent_id.name))
 
+    @api.constrains('parent_id', 'sequence', 'report_id')
+    def _validate_parent_sequence(self):
+        self.report_id._validate_parent_sequence()
```

Comment:
With this, you'll recheck the whole report once for each of its lines ; that's too many useless operations for what' you're trying to achieve.

Reply:
Anyway, have you read this ? https://github.com/odoo/odoo/pull/118341#pullrequestreview-1479703311 (it's the original PR introducing the constraint) 

My tests concluded this couldn't work on report lines. I think it's acceptable. What is your use case exactly ? Just lines created in data ?